velox/docs/functions/presto/conversion.rst

```diff
@@ -919,6 +979,54 @@ Invalid example
   SELECT cast('-3E+2.1' as decimal(12, 2)); -- Value is not a number
   SELECT cast('3E+' as decimal(12, 2)); -- Value is not a number
 
+Cast to IPAddress
+---------------
+
+From varchar type
+^^^^^^^^^^^^^^^^^
+
+To cast a varchar to IPAddress it must be in the form of either
+IPV4 or IPV6. It can be written in full form or shortened form.
+All IPV6 addresses that have the form of an IPV4 mapped IPV6
+address will be assumed to be IPV4 addresses. 
+
+Valid examples
+
+::
+
+  SELECT cast('2001:0db8:0000:0000:0000:ff00:0042:8329' as ipaddress); -- ipaddress '2001:db8::ff00:42:8329'
+  SELECT cast('1.2.3.4' as ipaddress); -- ipaddress '1.2.3.4'
+  SELECT cast('::ffff:ffff:ffff' as ipaddress); -- ipaddress '255.255.255.255'
+
+Invalid examples
+
+::
+
+  SELECT cast('2001:db8::1::1' as ipaddress); -- Invalid IP address
```

Comment:
> Invalid IP address

Is this the actual error reported?

Reply:
Yes, thats the error reported from folly's IPADDRESS parser.

https://github.com/facebook/folly/blob/2f7582dc07152c36ebd3959e9fd424ec2fb7667c/folly/IPAddressV6.cpp#L86C1-L119C1

https://github.com/facebook/folly/blob/2f7582dc07152c36ebd3959e9fd424ec2fb7667c/folly/IPAddress.cpp#L253